api/tbcapi/tbcapi.go

```diff
@@ -27,11 +30,40 @@ var (
 type (
 	PingRequest  protocol.PingRequest
 	PingResponse protocol.PingResponse
+	BtcHeader    struct {
+		Version uint32 `json:"version"`
+
+		// hex encoded byte array
+		PrevHash string `json:"prev_hash"`
+
+		// hex encoded byte array
+		MerkleRoot string `json:"merkle_root"`
+
+		Timestamp uint32 `json:"timestamp"`
+
+		// hex encoded int
+		Bits string `json:"bits"`
+
+		Nonce uint32 `json:"nonce"`
+	}
+	BtcBlockMetadata struct {
+		Height uint32    `json:"height"`
+		NumTx  uint32    `json:"num_tx"`
+		Header BtcHeader `json:"header"`
+	}
+	BtcBlockMetadataByNumRequest struct {
+		Height uint32 `json:"height"`
+	}
+	BtcBlockMetadataByNumResponse struct {
+		Block BtcBlockMetadata `json:"block"`
+	}
```

Comment:
I think we should each struct into a separate type block to help improve consistently with the rest of the code base.

Reply:
great idea, will update 👍